sympy/matrices/expressions/tests/test_matpow.py

```diff
@@ -70,7 +70,7 @@ def test_doit_square_MatrixSymbol_symsize():
 
 def test_doit_with_MatrixBase():
     X = ImmutableMatrix([[1, 2], [3, 4]])
-    assert MatPow(X, 0).doit() == ImmutableMatrix(Identity(2))
+    assert MatPow(X, 0).doit().as_explicit() == ImmutableMatrix(Identity(2))
```

Comment:
@moorepants 
I did it because ImmutableMatrix(Identity(2)) return the matrix in explicit form. (So i guess the test is wrong.)
So inspite of ImmutableMatrix(Identity(2)), it should be just Identity(2).
or we can just check the explicit form of LHS term